public/app/features/plugins/admin/components/VersionInstallButton.tsx

```diff
@@ -0,0 +1,110 @@
+import { css } from '@emotion/css';
+import { useEffect, useState } from 'react';
+import { gt } from 'semver';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, ConfirmModal, Spinner, useStyles2 } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+import { useInstall } from '../state/hooks';
+import { Version } from '../types';
+
+interface Props {
+  pluginId: string;
+  version: Version;
+  latestCompatibleVersion?: string;
+  installedVersion?: string;
+  disabled: boolean;
+  onClick: () => void;
+}
+
+export const VersionInstallButton = ({
+  pluginId,
+  version,
+  latestCompatibleVersion,
+  installedVersion,
+  disabled,
+  onClick,
```

Comment:
nit: since this component is really more than just a button you click I advice you to change this callback name to something else such as "onConfirmInstall" or something that tells is not just a click.